pkg/services/authz/legacy/client/client.go

```diff
@@ -0,0 +1,227 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc"
+)
+
+type CheckRequest struct {
+	// Namespace is either `org-<id>` or `stacks-<id>`.
+	Namespace string
+
+	// Subject is the typed identity we want to perform a check for
+	Subject string
+
+	// Action to check access for
+	Action string
+
+	// Resource is ~Kind eg dashboards
+	Resource string
+	// Attribute used to store permission
+	Attr string
+	// Name is the identifier for the resource.
+	// In grafana, this was historically called "UID", but in k8s, it is the name
+	Name string
+
+	// Contextuals are additional resource + name that should be checked.
+	// E.g. for dashboards this can be the folder that a it belong to.
+	Contextuals []Contextual
```

Comment:
In OpenFGA contextual tuples have almost the same definition as check request tuples. Because contextual is just additional relation tuples, like `folder:a - parent - dashboard:b`. Here we have only one part, like `folder:a read`. It's might be enough for our purpose, but we lose ability to specify relation, so we always should assume it's something like `parent` for folder. What do you think?

Reply:
Yeah I was thinking to make this a bit less generic and rename this to parents or something similar.

Reply:
Take a look at [ccbe39a](https://github.com/grafana/grafana/pull/94055/commits/ccbe39a0834433ed8cd80e034c71f02502ebcce4). I removed a lot of stuff that I was using for testing and only kept `Check` for now. I also changed from `Contextuals` to parents and they should be a list of uids and we always assume that they are folders. I am not sure we have any other case we need to cover.

The annoying part is that we require `Attr`. This will not be used when running this against open fga

Reply:
Yes, seems we only need parent as a contextual, so we can simplify it on our end.